packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -3913,9 +3913,8 @@ describe('NetworkController', () => {
                   params: ['latest', false],
                 },
                 response: {
-                  result: {
-                    baseFeePerGas: '0x1',
-                  },
+                  // first network is not EIP1559 compatible
```

Comment:
This test had a mistake in this test fixture; the test was assuming the first network wasn't EIP-1559 compatible, but the fixture returned a block that indicated that it was compatible. It only ever worked because `resetConnection` (the method under test here) was called before initialization was finished.

Reply:
Hmm. That seems like an extra detail this test should not be concerned with. I see that the test is waiting for `isEIP1559Compatible` to be changed after `resetConnection` is called, but now that I see this again, I'm not sure that's necessary. The code responsible for creating a new provider should be run synchronously.

What happens if you remove the call to `waitForStateChanges` and unwrap `resetConnection`?

Reply:
Great point! The test has been simplified here: https://github.com/MetaMask/core/pull/1180/commits/c8006ef2879b524f0602bbdd516d877cb8e7736b